_posts/2019-03-11-dask-nep18.md

```diff
@@ -0,0 +1,521 @@
+through compatible interface, we don't have to change the implementation at
+all, we simply pass a different array type as input, as simple as that.
+
+### Example with scikit-learn
+
+To demonstrate the ability we acquired, let's consider the following
+scikit-learn example (based on the original example
+[here](https://scikit-learn.org/stable/auto_examples/linear_model/plot_ols.html#sphx-glr-auto-examples-linear-model-plot-ols-py)):
+
+```python
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.linear_model import LinearRegression
+
+N = 1000
+
+# x from 0 to N
+x = N * np.random.random((40000, 1))
+
+# y = a*x + b with noise
+y = 0.5 * x + 1.0 + np.random.normal(size=x.shape)
+
+# create a linear regression model
+est = LinearRegression()
+```
+
+We can then fit the model,
+
+```python
+est.fit(x, y)
+```
+
+and obtain its time measurements:
+
+```
+CPU times: user 3.4 ms, sys: 0 ns, total: 3.4 ms
+Wall time: 2.3 ms
+```
+
+We can then use it for prediction on some test data,
+
+```python
+# predict y from the data
+x_new = np.linspace(0, N, 100)
+y_new = est.predict(x_new[:, np.newaxis])
+```
+
+And also check its time measurements:
+
+```python
+CPU times: user 1.16 ms, sys: 680 µs, total: 1.84 ms
+Wall time: 1.58 ms
+```
+
+And finally plot the results:
+
+```python
+# plot the results
+plt.figure(figsize=(4, 3))
+ax = plt.axes()
+ax.scatter(x, y, linewidth=3)
+ax.plot(x_new, y_new, color='black')
+
+ax.set_facecolor((1.0, 1.0, 0.42))
+
+ax.set_xlabel('x')
+ax.set_ylabel('y')
+
+ax.axis('tight')
+
+plt.show()
+```
+
+<blockquote class="imgur-embed-pub"
+            lang="en"
+            data-id="a/guL31Hb">
+<a href="//imgur.com/a/guL31Hb"></a>
+</blockquote>
+<script async src="//s.imgur.com/min/embed.js" charset="utf-8"></script>
+
+
+### Example with Dask-GLM
+
+The only thing we have to change from the code before is the first block, where
+we import libraries and create arrays:
+
+```python
+import numpy as np
+from dask_glm.estimators import LinearRegression
+import matplotlib.pyplot as plt
+
+N = 1000
+
+# x from 0 to N
+x = N * np.random.random((40000, 1))
+
+# y = a*x + b with noise
+y = 0.5 * x + 1.0 + np.random.normal(size=x.shape)
+
+# create a linear regression model
+est = LinearRegression(solver='lbfgs')
+```
+
+The rest of the code and also the plot look alike the previous scikit-learn
+example, so we're ommitting those here for brevity. Note also that we could
+have called `LinearRegression()` without any arguments, but for this example
+we chose the
+[`lbfgs`](https://docs.scipy.org/doc/scipy/reference/optimize.minimize-lbfgsb.html)
+solver, that converges reasonably fast.
+
+We can also have a look at the timing results for fitting, followed by those
+for predicting with Dask-GLM:
+
+```
+# Fitting
+CPU times: user 9.66 ms, sys: 116 µs, total: 9.78 ms
+Wall time: 8.94 ms
+
+# Predicting
+CPU times: user 130 µs, sys: 327 µs, total: 457 µs
+Wall time: 1.06 ms
+```
```

Comment:
Would it be possible to increase the size of the input a bit?  I think that dask-glm is only really relevant once we get to larger sizes.  

Also, I recommend using `LogisticRegression` rather than `LinearRegression` if that's an easy change to make.  I suspect that seeing something other than `LinearRegression` will be more motivating.

Reply:
> Would it be possible to increase the size of the input a bit? I think that dask-glm is only really relevant once we get to larger sizes.

Unfortunately, not for fitting, as we run out of memory (I didn't want to dive deep now to check in detail why). I didn't try that for prediction though. I'll check.

> Also, I recommend using `LogisticRegression` rather than `LinearRegression` if that's an easy change to make. I suspect that seeing something other than `LinearRegression` will be more motivating.

I will check if it's possible without too big changes.

Reply:
The results from the new graphs using `LogisticRegression`.